cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerConfigService.java

```diff
@@ -82,23 +85,35 @@ public void disableKnoxAutorestartIfCmVersionAtLeast(Versioned versionAtLeast, A
     }
 
     private void disableKnoxAutorestart(ApiClient client, String clusterName) {
-        try {
             ServicesResourceApi servicesResourceApi = clouderaManagerApiFactory.getServicesResourceApi(client);
-            String knoxServiceName = getKnoxServiceName(clusterName, servicesResourceApi);
+            getKnoxServiceName(clusterName, servicesResourceApi)
+                    .ifPresentOrElse(
+                            doDisableKnoxAutorestart(clusterName, servicesResourceApi),
+                            () -> LOGGER.info("KNOX service name is missing, skipping disabling the autorestart property."));
+    }
 
+    private Consumer<String> doDisableKnoxAutorestart(String clusterName, ServicesResourceApi servicesResourceApi) {
+        return knoxServiceName -> {
             ApiConfig autorestartConfig = new ApiConfig().name(KNOX_AUTORESTART_ON_STOP).value(Boolean.FALSE.toString());
             ApiServiceConfig serviceConfig = new ApiServiceConfig().addItemsItem(autorestartConfig);
-            servicesResourceApi.updateServiceConfig(clusterName, knoxServiceName, "", serviceConfig);
-        } catch (ApiException e) {
-            LOGGER.debug("Failed to set autorestart_on_stop to KNOX in Cloudera Manager.", e);
-        }
+            try {
+                servicesResourceApi.updateServiceConfig(clusterName, knoxServiceName, "", serviceConfig);
+            } catch (ApiException e) {
+                LOGGER.debug("Failed to set autorestart_on_stop to KNOX in Cloudera Manager.", e);
+            }
+        };
     }
 
-    private String getKnoxServiceName(String clusterName, ServicesResourceApi servicesResourceApi) throws ApiException {
-        ApiServiceList serviceList = servicesResourceApi.readServices(clusterName, DataView.SUMMARY.name());
-        return serviceList.getItems().stream()
-                .filter(service -> KNOX_SERVICE.equals(service.getType()))
-                .map(ApiService::getName)
-                .findFirst().orElse(KNOX_SERVICE);
+    private Optional<String> getKnoxServiceName(String clusterName, ServicesResourceApi servicesResourceApi) {
+        try {
+            ApiServiceList serviceList = servicesResourceApi.readServices(clusterName, DataView.SUMMARY.name());
+            return serviceList.getItems().stream()
+                    .filter(service -> KNOX_SERVICE.equals(service.getType()))
+                    .map(ApiService::getName)
+                    .findFirst();
+        } catch (ApiException e) {
+            LOGGER.debug("Failed to get KNOX service name from Cloudera Manager.", e);
```

Comment:
Maybe the cluster name should be added to the log message

Reply:
Ideally the log entry should already contain it.